lcd.lua

```diff
@@ -128,16 +128,34 @@ local clearscreen = function(pos)
 	end
 end
 
+local set_texture = function(ent)
+	local meta = minetest.get_meta(ent.object:getpos())
+	local text = meta:get_string("text")
+	ent.object:set_properties({textures={generate_texture(create_lines(text))}})
+end
+
 local prepare_writing = function(pos)
-	local lcd_info = lcds[minetest.get_node(pos).param2]
-	if lcd_info == nil then return end
-	local text = minetest.add_entity(
-		{x = pos.x + lcd_info.delta.x,
-		 y = pos.y + lcd_info.delta.y,
-		 z = pos.z + lcd_info.delta.z}, "digilines_lcd:text")
-	text:setyaw(lcd_info.yaw or 0)
-	--* text:setpitch(lcd_info.yaw or 0)
-	return text
+	local existing
+	local objects = minetest.get_objects_inside_radius(pos, 0.5)
+	for _, o in ipairs(objects) do
+		local o_entity = o:get_luaentity()
+		if o_entity and o_entity.name == "digilines_lcd:text" then
+			existing = o_entity
+			break
+		end
+	end
+	if not existing then
+		local lcd_info = lcds[minetest.get_node(pos).param2]
+		if lcd_info == nil then return end
```

Comment:
Split up into multiple lines
```suggestion
		if not lcd_info then
			return
		end
```